### PR TITLE
Do not fail input checks for endpoints with .amazonaws.com as prefix

### LINKED ISSUE #570: minio-py fails erroneously with minio.error.InvalidEndpointError: InvalidEndpointError: message: Amazon S3 hostname should be s3.amazonaws.com
I have minio installed to our kubernetes cluster on amazon. The hostname ends in amazonaws.com but is not S3. When I run the minio-py with the correct credentials, the code fails here:

https://github.com/minio/minio-py/blob/master/minio/helpers.py#L276-L279

The check for s3 ignores the fact that I may have an external IP exposed through amazon that is not s3, but rather minio - in this case hosted on our kubernetes cluster.

Do you need a pull request?

### PR DESCRIPTION
**Pull request by harshavardhana**
Additionally also support extracting region from the endpoint, this
technique is an optimization avoids GetBucketLocation() request
and makes it more performant.

Fixes #570

### CODE REVIEW

File: minio/helpers.py [hunk 1]
```diff
@@ -49,7 +49,24 @@
 _ALLOWED_HOSTNAME_REGEX = re.compile(
     '^((?!-)[A-Z\\d-]{1,63}(?<!-)\\.)*((?!-)[A-Z\\d-]{1,63}(?<!-))$',
     re.IGNORECASE)
+_EXTRACT_REGION_REGEX = re.compile('s3-(.+?).amazonaws.com')
```
Comment (poornas) [inline minio/helpers.py]: regex wont work if endpoint is like s3.us-west-1.amazonaws.com

**Comment by harshavardhana** [comment]
@donatello need your reviews.

**Comment by harshavardhana** [comment]
@donatello pending your review.

File: minio/helpers.py [hunk 2]
```diff
@@ -49,7 +49,24 @@
 _ALLOWED_HOSTNAME_REGEX = re.compile(
     '^((?!-)[A-Z\\d-]{1,63}(?<!-)\\.)*((?!-)[A-Z\\d-]{1,63}(?<!-))$',
     re.IGNORECASE)
+_EXTRACT_REGION_REGEX = re.compile('s3[.-]?(.+?).amazonaws.com')
```
Comment (donatello) [inline minio/helpers.py]: There is one more variant that seems easy to add here e.g. `s3.dualstack.us-east-2.amazonaws.com` - see https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region

Comment (harshavardhana) [reply minio/helpers.py]: done

**Comment by harshavardhana** [comment]
@poornas need your review again.

**Pull request merged**